internal/registry/registry.go

```diff
@@ -23,22 +23,48 @@ import (
 	"github.com/elastic/go-sysinfo/types"
 )
 
-var (
-	hostProvider    HostProvider
-	processProvider ProcessProvider
-)
+type HostOptsCreator = func(ProviderOptions) HostProvider
+type ProcessOptsCreator = func(ProviderOptions) ProcessProvider
 
+// HostProvider defines interfaces that provide host-specific metrics
 type HostProvider interface {
 	Host() (types.Host, error)
 }
 
+// ProcessProvider defines interfaces that provide process-specific metrics
 type ProcessProvider interface {
 	Processes() ([]types.Process, error)
 	Process(pid int) (types.Process, error)
 	Self() (types.Process, error)
 }
 
+type ProviderOptions struct {
+	Hostfs string
+}
+
+var (
+	hostProvider            HostProvider
+	processProvider         ProcessProvider
+	processProviderWithOpts ProcessOptsCreator
+	hostProviderWithOpts    HostOptsCreator
+)
+
+// Register a metrics provider. `provider` should implement one or more of `ProcessProvider` or `HostProvider`
 func Register(provider interface{}) {
+	if h, ok := provider.(ProcessOptsCreator); ok {
+		if processProviderWithOpts != nil {
+			panic(fmt.Sprintf("ProcessOptsCreator already registered"))
+		}
+		processProviderWithOpts = h
+	}
+
+	if h, ok := provider.(HostOptsCreator); ok {
+		if processProviderWithOpts != nil {
+			panic(fmt.Sprintf("ProcessOptsCreator already registered"))
```

Comment:
I think this is mean to be checking against `hostProviderWithOpts`.

Can you make the messages consistent with the existing format (it includes the already registered provider) or vice-versa.